#VRML_SIM V6.0 utf8
WorldInfo {
  info [
    "Demonstration of the alternate tripod gait using linear and rotational servos."
    "Author: Yvan Bourquin"
    "Biologically Inspired Robotics Group (BIRG)"
    "Date: 22 September 2005"
  ]
  title "Hexapod"
  CFM 0.0001
  ERP 0.6
  basicTimeStep 8
  displayRefresh 4
  runRealTime TRUE
}
Viewpoint {
  orientation -0.0128766 0.996972 -0.0766942 1.73508
  position 1.17273 0.0880683 0.0960653
  follow "simpleRHex"
}
Background {
}
DEF GROUND Solid {
  translation -50 0 -50
  children [
    Shape {
      geometry ElevationGrid {
        color Color {
          color [
            0.713725 0 0
            0.184314 0.184314 0.184314
          ]
        }
        xDimension 50
        xSpacing 2
        zDimension 50
        zSpacing 2
      }
    }
  ]
  boundingObject IndexedFaceSet {
    coord Coordinate {
      point [
        100 0 100
        100 0 0
        0 0 0
        0 0 100
      ]
    }
    coordIndex [
      0, 1, 2, 3, -1
    ]
  }
  locked TRUE
}
DEF HEXAPOD Robot {
  translation 0 0.06 1
  rotation -2.45508e-005 1 3.39756e-005 1.30267
  children [
    DEF HIP_SERVO_R0 Servo {
      translation 0.2 0 0.05
      rotation 0 0 1 0
      children [
        DEF HIP_SHAPE Shape {
          appearance DEF HIP_APPEARANCE Appearance {
            material Material {
              diffuseColor 1 0 0
            }
          }
          geometry Cylinder {
            height 0.06
            radius 0.02
          }
        }
        DEF FEMUR_TRANS Transform {
          translation 0 0 0.06
          children [
            DEF FEMUR_SHAPE Shape {
              appearance DEF METAL_APPEARANCE Appearance {
                material Material {
                  diffuseColor 0.5 0.5 0.5
                }
              }
              geometry Box {
                size 0.02 0.02 0.1
              }
            }
          ]
        }
        DEF KNEE_TRANS Transform {
          translation 0 0 0.11
          children [
            DEF KNEE_SHAPE Shape {
              appearance DEF KNEE_APPEARANCE Appearance {
                material Material {
                  diffuseColor 0 0.6 0.2
                }
              }
              geometry Cylinder {
                height 0.06
                radius 0.02
              }
            }
          ]
        }
        DEF KNEE_SERVO_R0 Servo {
          translation 0 -0.03 0.11
          rotation 1 0 0 1.5708
          children [
            DEF TIBIA_TRANS Transform {
              rotation 1 0 0 1.5708
              children [
                DEF TIBIA_SHAPE Shape {
                  appearance USE METAL_APPEARANCE
                  geometry Cylinder {
                    height 0.06
                    radius 0.01
                  }
                }
              ]
            }
          ]
          name "knee_servo_r0"
          boundingObject USE TIBIA_TRANS
          physics DEF STANDARD_PHYSICS Physics {
            coulombFriction 0.5
          }
          type "linear"
          maxForce 25
          position -0.000102829
          minPosition -0.02
          maxPosition 0.02
          minStop -0.02
          maxStop 0.02
        }
      ]
      name "hip_servo_r0"
      boundingObject USE KNEE_TRANS
      physics USE STANDARD_PHYSICS
      maxVelocity 5
      position 63.6172
    }
    DEF HIP_SERVO_R1 Servo {
      translation 0 0 0.05
      rotation 0 0 1 0
      children [
        USE HIP_SHAPE
        USE FEMUR_TRANS
        USE KNEE_TRANS
        DEF KNEE_SERVO_R1 Servo {
          translation 0 -0.03 0.11
          rotation 1 0 0 1.5708
          children [
            USE TIBIA_TRANS
          ]
          name "knee_servo_r1"
          boundingObject USE TIBIA_TRANS
          physics USE STANDARD_PHYSICS
          type "linear"
          maxForce 25
          position -8.71939e-005
          minPosition -0.02
          maxPosition 0.02
          minStop -0.02
          maxStop 0.02
        }
      ]
      name "hip_servo_r1"
      boundingObject USE KNEE_TRANS
      physics USE STANDARD_PHYSICS
      maxVelocity 5
      position 62.0438
    }
    DEF HIP_SERVO_R2 Servo {
      translation -0.2 0 0.05
      rotation 0 0 1 0
      children [
        USE HIP_SHAPE
        USE FEMUR_TRANS
        USE KNEE_TRANS
        DEF KNEE_SERVO_R2 Servo {
          translation 0 -0.03 0.11
          rotation 1 0 0 1.5708
          children [
            USE TIBIA_TRANS
          ]
          name "knee_servo_r2"
          boundingObject USE TIBIA_TRANS
          physics USE STANDARD_PHYSICS
          type "linear"
          maxForce 25
          position -0.000104959
          minPosition -0.02
          maxPosition 0.02
          minStop -0.02
          maxStop 0.02
        }
      ]
      name "hip_servo_r2"
      boundingObject USE KNEE_TRANS
      physics USE STANDARD_PHYSICS
      maxVelocity 5
      position 63.6172
    }
    DEF HIP_SERVO_L0 Servo {
      translation 0.2 0 -0.05
      rotation 0 0 1 0
      children [
        USE HIP_SHAPE
        DEF LEFT_FEMUR_TRANS Transform {
          translation 0 0 -0.06
          children [
            USE FEMUR_SHAPE
          ]
        }
        DEF LEFT_KNEE_TRANS Transform {
          translation 0 0 -0.11
          children [
            USE KNEE_SHAPE
          ]
        }
        DEF KNEE_SERVO_L0 Servo {
          translation 0 -0.03 -0.11
          rotation 1 0 0 1.5708
          children [
            USE TIBIA_TRANS
          ]
          name "knee_servo_l0"
          boundingObject USE TIBIA_TRANS
          physics USE STANDARD_PHYSICS
          type "linear"
          maxForce 25
          position -8.11399e-005
          minPosition -0.02
          maxPosition 0.02
          minStop -0.02
          maxStop 0.02
        }
      ]
      name "hip_servo_l0"
      boundingObject USE LEFT_KNEE_TRANS
      physics USE STANDARD_PHYSICS
      maxVelocity 5
      position -62.0438
    }
    DEF HIP_SERVO_L1 Servo {
      translation 0 0 -0.05
      rotation 0 0 1 0
      children [
        USE HIP_SHAPE
        USE LEFT_FEMUR_TRANS
        USE LEFT_KNEE_TRANS
        DEF KNEE_SERVO_L1 Servo {
          translation 0 -0.03 -0.11
          rotation 1 0 0 1.5708
          children [
            USE TIBIA_TRANS
          ]
          name "knee_servo_l1"
          boundingObject USE TIBIA_TRANS
          physics USE STANDARD_PHYSICS
          type "linear"
          maxForce 25
          position -0.000139654
          minPosition -0.02
          maxPosition 0.02
          minStop -0.02
          maxStop 0.02
        }
      ]
      name "hip_servo_l1"
      boundingObject USE LEFT_KNEE_TRANS
      physics USE STANDARD_PHYSICS
      maxVelocity 5
      position -63.6173
    }
    DEF HIP_SERVO_L2 Servo {
      translation -0.2 0 -0.05
      rotation 0 0 1 0
      children [
        USE HIP_SHAPE
        USE LEFT_FEMUR_TRANS
        USE LEFT_KNEE_TRANS
        DEF KNEE_SERVO_L2 Servo {
          translation 0 -0.03 -0.11
          rotation 1 0 0 1.5708
          children [
            USE TIBIA_TRANS
          ]
          name "knee_servo_l2"
          boundingObject USE TIBIA_TRANS
          physics USE STANDARD_PHYSICS
          type "linear"
          maxForce 25
          position -7.89554e-005
          minPosition -0.02
          maxPosition 0.02
          minStop -0.02
          maxStop 0.02
        }
      ]
      name "hip_servo_l2"
      boundingObject USE LEFT_KNEE_TRANS
      physics USE STANDARD_PHYSICS
      maxVelocity 5
      position -62.0438
    }
    DEF BODY_GROUP Group {
      children [
        DEF BODY_SHAPE Shape {
          appearance USE METAL_APPEARANCE
          geometry Box {
            size 0.48 0.05 0.1
          }
        }
      ]
    }
    PointLight {
      attenuation 0 0 1
      intensity 11
      location 0 3.5 0
      castShadows TRUE
    }
    PointLight {
      attenuation 0 0 1
      intensity 11
      location 1 3.5 -1
    }
  ]
  name "hexapod"
  boundingObject USE BODY_SHAPE
  physics USE STANDARD_PHYSICS
  controller "doNothing"
}
DEF SimpleRHex Robot {
  translation 0 0.16 0
  rotation 0 0 1 0
  children [
    PointLight {
      attenuation 0 0 1
      intensity 11
      location 1 3.5 -1
    }
    DEF LEG_L0 Servo {
      translation 0.12 0 0.16
      rotation 1 0 0 1.5708
      children [
        DEF LEG_TRANSFORM Transform {
          translation 0 0 0.07
          children [
            DEF LEG Shape {
              geometry Box {
                size 0.01 0.01 0.13
              }
            }
          ]
        }
      ]
      name "servo_l0"
      boundingObject USE LEG_TRANSFORM
      physics DEF MY_PHYSICS Physics {
      }
    }
    DEF LEG_L1 Servo {
      translation 0.18 0 0
      rotation 1 0 0 1.5708
      children [
        USE LEG_TRANSFORM
      ]
      name "servo_l1"
      boundingObject USE LEG_TRANSFORM
      physics USE MY_PHYSICS
    }
    DEF LEG_L2 Servo {
      translation 0.12 0 -0.16
      rotation 1 0 0 1.5708
      children [
        USE LEG_TRANSFORM
      ]
      name "servo_l2"
      boundingObject USE LEG_TRANSFORM
      physics USE MY_PHYSICS
    }
    DEF LEG_R0 Servo {
      translation -0.12 0 0.16
      rotation 1 0 0 1.5708
      children [
        USE LEG_TRANSFORM
      ]
      name "servo_r0"
      boundingObject USE LEG_TRANSFORM
      physics USE MY_PHYSICS
    }
    DEF LEG_R1 Servo {
      translation -0.18 0 0
      rotation 1 0 0 1.5708
      children [
        USE LEG_TRANSFORM
      ]
      name "servo_r1"
      boundingObject USE LEG_TRANSFORM
      physics USE MY_PHYSICS
    }
    DEF LEG_R2 Servo {
      translation -0.12 0 -0.16
      rotation 1 0 0 1.5708
      children [
        USE LEG_TRANSFORM
      ]
      name "servo_r2"
      boundingObject USE LEG_TRANSFORM
      physics USE MY_PHYSICS
    }
    DEF THE_BODY Group {
      children [
        DEF CAMERA Transform {
          translation 0 0 -0.22
          rotation 1 0 0 1.56
          children [
            Shape {
              geometry Cylinder {
                height 0.06
                radius 0.02
              }
            }
          ]
        }
        DEF MID Transform {
          rotation 0 0 1 1.56
          children [
            DEF MID_AXIS Shape {
              geometry Cylinder {
                height 0.35
                radius 0.025
              }
            }
          ]
        }
        DEF BACK Transform {
          translation 0 0 -0.16
          rotation 0 0 1 1.56
          children [
            DEF AXIS Shape {
              geometry Cylinder {
                height 0.23
                radius 0.025
              }
            }
          ]
        }
        DEF FRONT Transform {
          translation 0 0 0.16
          rotation 0 0 1 1.56
          children [
            DEF AXIS Shape {
              geometry Cylinder {
                height 0.23
                radius 0.025
              }
            }
          ]
        }
        DEF MID_BOX Shape {
          geometry Box {
            size 0.32 0.09 0.07
          }
        }
        DEF HEAD Transform {
          translation -3.46945e-018 0 0.2
          rotation 0 0 1 1.56
          children [
            DEF HEAD_TAIL Shape {
              geometry Cylinder {
                height 0.19
                radius 0.04
              }
            }
          ]
        }
        DEF TAIL Transform {
          translation 0 0 -0.2
          rotation 0 0 1 1.56
          children [
            DEF HEAD_TAIL Shape {
              geometry Cylinder {
                height 0.19
                radius 0.04
              }
            }
          ]
        }
        DEF MAIN_BOX Shape {
          geometry Box {
            size 0.2 0.1 0.4
          }
        }
      ]
    }
  ]
  name "simpleRHex"
  boundingObject USE MAIN_BOX
  physics Physics {
    mass 100
  }
  controller "simpleRhex"
}
PointLight {
  attenuation 0 0 1
  intensity 11
  location 1 3.5 -1
}
DEF obstacle Solid {
  translation -0.267595 0.11 1.3419
  children [
    DEF OBSTACLE_BOX Shape {
      geometry Box {
        size 0.2 0.08 0.2
      }
    }
  ]
  boundingObject USE OBSTACLE_BOX
  physics Physics {
  }
}
USE obstacle
